File: sandbox/clib/tests/arr.cpp
#include <iostream>
#define MAX 2000000
using namespace std;

int main(int argc, char *argv[]) {
    int a[MAX] = {1};

    for (int i = 0; i < MAX; i++) {
        a[i] = i;
    }

    cout << a[MAX-1] << " " << sizeof(a) << endl;

    return 0;
}

File: sandbox/clib/tests/Arr.java
package tests;

public class Arr {
    public static final int MAX = 20000000;

    public static void main(String[] args) {
        int[] a = new int[MAX];

        for (int i = 0; i < MAX; i++) {
            a[i] = i;
        }

        System.out.println(a[MAX-1]);
    }
}
